consensus/hotstuff_leader.go

```diff
@@ -51,7 +49,7 @@ func (handler *HotstuffLeaderMessageHandler) HandleNewRoundMessage(m *consensusM
 
 	// Likely to be `nil` if blockchain is progressing well.
 	// TECHDEBT: How do we properly validate `prepareQC` here?
-	highPrepareQC := m.findHighQC(m.messagePool[NewRound])
+	highPrepareQC := m.findHighQC(m.consensusMessagePool[NewRound].GetAll()) //TODO: improve this
```

Comment:
Lemme say it out loud, perhaps it's a new ticket straight away.

I was thinking to further optimize this by having an abstraction around/extension on the current logic so that every time a new message is added to the mempool, it's immediately checked if it's a highPrepareQC and in that case that information is "memoized".

This way, we don't have to scan all the messages in the mempool to get the HighQC, we would have it already.

👍new ticket with the above description
👎small enough to add scope here

Curious about your thoughts on this. Maybe I am optimizing prematurely

Reply:
Let's rename `DISCUSS` to `CONSIDERATION(M5)`. 

This approach is basically how Hotstuff specs out the default design and you're providing a good implementation optimization. Low priority enough for now to not need a ticket.